source/common/network/address_impl.cc

```diff
@@ -300,8 +353,40 @@ PipeInstance::PipeInstance(const std::string& pipe_path, mode_t mode,
   pipe_.mode_ = mode;
 }
 
+PipeInstance::PipeInstance(absl::Status& error, const sockaddr_un* address, socklen_t ss_len,
+                           mode_t mode, const SocketInterface* sock_interface)
+    : InstanceBase(Type::Pipe, sockInterfaceOrDefault(sock_interface)) {
+  if (address->sun_path[0] == '\0') {
+#if !defined(__linux__)
+    error = absl::FailedPreconditionError("Abstract AF_UNIX sockets are only supported on linux.");
```

Comment:
I think you can drop this ifdef block as you will repeat the same test in initHelper.

Reply:
It is not exactly the same test...

Reply:
ah you're right. sorry about that!